packages/spotlight/bin/instrument.js

```diff
@@ -1,4 +1,56 @@
-import { consoleLoggingIntegration, init } from "@sentry/node";
+import { consoleLoggingIntegration, getClient, init } from "@sentry/node";
+import { parseCLIArgs } from "@spotlightjs/sidecar/cli";
+import { DEFAULT_PORT } from "@spotlightjs/sidecar/constants";
+
+// Parse CLI arguments to determine the port this Spotlight instance will use
+const { port: instancePort } = parseCLIArgs();
+
+// Check if SENTRY_SPOTLIGHT points to this instance
+const spotlightEnv = process.env.SENTRY_SPOTLIGHT;
+let disableSpotlight = false;
+
+// Note: If port is set to 0 (dynamic port assignment), we cannot detect the actual port
+// before starting the server, so we cannot prevent the feedback loop in that case.
+// The server will be assigned a port by the OS after it starts, and if SENTRY_SPOTLIGHT
+// happens to point to that same port, a feedback loop may occur.
+if (spotlightEnv && instancePort !== 0) {
+  let targetPort;
+  let targetHost;
+
+  // SENTRY_SPOTLIGHT can be:
+  // 1. A full URL like "http://localhost:8969"
+  // 2. A truthy value (true, t, y, yes, on, 1) which means use the default URL
+  const TRUTHY_ENV_VALUES = new Set(["true", "t", "y", "yes", "on", "1"]);
```

Comment:
i don't know how  SENTRY_SPOTLIGHT can be all those things but ok

Reply:
https://github.com/getsentry/sentry-javascript/blob/98de756506705b60d1ca86cbbcfad3fd76062f8f/packages/node-core/src/utils/envToBool.ts#L2